velox/exec/tests/TaskTest.cpp

```diff
@@ -581,4 +597,46 @@ TEST_F(TaskTest, singleThreadedCrossJoin) {
   }
 }
 
+// Check if a downstream task can elegantly close all related upstream tasks
+// once the downstream task completes/aborts.
+TEST_F(TaskTest, checkExchangeSourceClosedAfterAbort) {
+  // Create and start a "remote" task
+  std::string remoteTaskId = "mock://task-0";
+  auto remoteTaskPlan =
+      exec::test::PlanBuilder()
+          .tableScan(ROW({"c0", "c1"}, {INTEGER(), VARCHAR()}))
+          .partitionedOutput({}, 1)
+          .planFragment();
+  auto remoteTask = makeTask(remoteTaskId, remoteTaskPlan, 0);
+  remoteTask->start(remoteTask, 1, 5);
+
+  // Create and start a new task consuming the output of the remote task
+  auto plan = exec::test::PlanBuilder()
+                  .exchange(remoteTaskPlan.planNode->outputType())
+                  .partitionedOutput({}, 1)
+                  .planFragment();
+  plan.executionStrategy = core::ExecutionStrategy::kGrouped;
```

Comment:
Is it important to use grouped execution? It is a bit strange to apply it to a non-leaf stage.